src/components/inputs/TokenInput/TokenInput.vue

```diff
@@ -246,14 +255,14 @@ const setMax = () => {
     inputAlignRight
     @blur="emit('blur', $event)"
     @input="emit('input', $event)"
-    @update:model-value="handleAmountChange($event)"
+    @update:model-value="emit('update:amount', $event)"
```

Comment:
Overflow protection is handled by BalTextInput itself so we shouldn't go through `changeAmount` function here and instead just emit event.